web/src/main/java/se/inera/intyg/webcert/web/service/facade/impl/CertificateTypeMessageServiceImpl.java

```diff
@@ -0,0 +1,169 @@
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import se.inera.intyg.infra.security.authorities.AuthoritiesHelper;
+import se.inera.intyg.infra.security.common.model.AuthoritiesConstants;
+import se.inera.intyg.schemas.contract.Personnummer;
+import se.inera.intyg.webcert.web.service.facade.CertificateTypeMessageService;
+import se.inera.intyg.webcert.web.service.user.WebCertUserService;
+import se.inera.intyg.webcert.web.service.utkast.UtkastService;
+import se.inera.intyg.webcert.web.service.utkast.dto.PreviousIntyg;
+
+@Service
+public class CertificateTypeMessageServiceImpl implements CertificateTypeMessageService {
+
+    private final UtkastService utkastService;
+    private final WebCertUserService webCertUserService;
+    private AuthoritiesHelper authoritiesHelper;
+
+    @Autowired
+    public CertificateTypeMessageServiceImpl(UtkastService utkastService,
+        WebCertUserService webCertUserService, AuthoritiesHelper authoritiesHelper) {
+        this.utkastService = utkastService;
+        this.webCertUserService = webCertUserService;
+        this.authoritiesHelper = authoritiesHelper;
+    }
+
+    @Override
+    public Optional<String> get(String certificateType, Personnummer patientId) {
+        if (!authoritiesHelper.isFeatureActive(AuthoritiesConstants.FEATURE_UNIKT_INTYG, certificateType)
+            && !authoritiesHelper.isFeatureActive(AuthoritiesConstants.FEATURE_UNIKT_INTYG_INOM_VG, certificateType)) {
+            return Optional.empty();
+        }
+        final var user = webCertUserService.getUser();
+        final Map<String, Map<String, PreviousIntyg>> previousCertificates = utkastService
+            .checkIfPersonHasExistingIntyg(patientId, user, null);
+
+        final var previousCertificateMap = previousCertificates.getOrDefault(INTYG_INDICATOR, Collections.emptyMap());
+        if (previousCertificateMap.containsKey(certificateType)) {
+            final var previousIntyg = previousCertificateMap.get(certificateType);
+            if (certificateTypeIsDb(certificateType)) {
+                return getOptionalStringCertificateDb(previousIntyg);
+            } else {
+                return getOptionalStringCertificateDoi(previousIntyg);
+            }
+        }
+
+        final var previousDraftMap = previousCertificates.getOrDefault(UTKAST_INDICATOR, Collections.emptyMap());
+        if (previousDraftMap.containsKey(certificateType)) {
+            final var previousIntyg = previousDraftMap.get(certificateType);
+            if (certificateTypeIsDb(certificateType)) {
+                return getOptionalStringDraftDb(previousIntyg);
+            } else {
+                return getOptionalStringDraftDoi(previousIntyg);
+            }
+        }
+
+        return Optional.empty();
+    }
+
+    private static Optional<String> getOptionalStringCertificateDoi(final PreviousIntyg previousIntyg) {
```

Comment:
For consistency you could remove the static-modifier. No need to have the methods as static, but it doesn't hur either.